### PR TITLE
refactor(multi_object_tracker): rework parameters

### PR DESCRIPTION
**Pull request by tby-udel**
## Description
Implement the ROS Node configuration layout described in https://github.com/orgs/autowarefoundation/discussions/3371 for the `multi_object_tracker` package.

 * Remove the default value from the source code in order to ensure all parameter values are passed from the parameter files.
 * Use types that match the ones returned from declare_parameter() in order to prevent confusion and potential coding errors.
 * Move the default parameter file to a standardized location. Use the JSON Schema for that.
 * Updated readme file.

## Tests performed
Package built and launched locally.

## Notes for reviewers
 Nothing inside the src code part was changed, only taking the parameters out.

## Interface changes
Parameter values are now required to be passed to the node when launched.

## Effects on system behavior
None

## Pre-review checklist for the PR author
The PR author **must** check the checkboxes below when creating the PR.

 * [x]  I've confirmed the [contribution guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/).
 * [x]  The PR follows the [pull request guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/).

 ## In-review checklist for the PR reviewers
 The PR reviewers **must** check the checkboxes below before approval.

 * [ ]  The PR follows the [pull request guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/).
 * [ ]  The PR has been properly tested.
 * [ ]  The PR has been reviewed by the code owners.
 
 ## Post-review checklist for the PR author
 The PR author **must** check the checkboxes below before merging.
 
 * [ ]  There are no open discussions or they are tracked via tickets.
 * [ ]  The PR is ready for merge.
 
 After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

**Review by yukkysaito — APPROVED** [review]
LGTM

**Comment by mitsudome-r** [comment]
@tby-udel Thanks for the PR. 

However, the PR seems to be failing in the CI for DCO check. When you push commits, could you push with signed-off commits?

For example by running the following command:
`git rebase -i --signoff origin/main`

**Comment by tby-udel** [comment]
Thank you for your reminding, I have pushed the signed-off version.

**Review by technolojin — APPROVED** [review]
LGTM
Thank you for your contribution.

File: perception/lidar_centerpoint/schema/centerpoint_base.schemal.json
Comment (mitsudome-r) [inline perception/lidar_centerpoint/schema/centerpoint_base.schemal.json:1]: There is a typo in the file name. It should be `schema` instead of `schemal`

Comment (tby-udel) [reply perception/lidar_centerpoint/schema/centerpoint_base.schemal.json:1]: Corrected.

**Comment by technolojin** [comment]
Let me update the branch and merge to the main, since the author @tby-udel did not responded for a long time.

**Comment by technolojin** [comment]
@kminoda @knzo25 can you check the 'lidar_centerpoint' part? I do not know about `*_base.param.yaml` configs.

**Comment by tby-udel** [comment]
I have pushed the signoff version, what should I do next?

**Comment by technolojin** [comment]
> I have pushed the signoff version, what should I do next?

The checker says there are mismatch between /config/centerpoint_base.param.yaml 🆚 schema/centerpoint_base.schema.json
https://github.com/autowarefoundation/autoware.universe/actions/runs/9820269981/job/27264059291?pr=7333

**Comment by tby-udel** [comment]
I've reverted the changes in the lidar_centerpoint section to match the current version in autoware.universe.

**Comment by technolojin** [comment]
@tby-udel 
There are conflicts in multi_object_tracker.

**Comment by tby-udel** [comment]
I solved the conflicts in schema files, and I didn't find conflicts in these src/tracker/model/*.cpp files.

**Comment by technolojin** [comment]
@tby-udel 
This package had some PRs after this PR is opened and one of them was over many files.
https://github.com/autowarefoundation/autoware.universe/pull/7863

Let me try to update this PR and resolve the conflict.

Screenshot from 2024-07-16 11-31-27

**Comment by technolojin** [comment]
@tby-udel Merge is ready! Please go on your timing.

**Comment by tby-udel** [comment]
@technolojin Thank you so much for your help. I have merged branch 'main' into multi-object-tracker-branch, and it seems this PR is still waiting for some check before it can be merged. Working on OpenAD Kit is my first time experience on github, could you please tell me if this is the case? Or are there any other things that I should do on this PR?

**Comment by technolojin** [comment]
@tby-udel 

1. You may not need to merge the main branch unless there is a big difference. At the moment I checked yesterday was fine (no need to merge the main). The last thing to finish the work is to press `Squash and Merge` button.

2. Now, I approved to run the tests. It need to be tested after the branch is updated. You can press `Enable auto-merge (squash)` now, so that the final work can be done as soon as the test is passed. 
Screenshot from 2024-07-17 09-09-21

**Comment by technolojin** [comment]
@tby-udel 
Now all the conditions are clear.
image

You can press `Squash and merge` button on the left bottom side, and then it will be merged!

**Comment by tby-udel** [comment]
image
@technolojin Thank you very much for your instructions! However, I checked everywhere on this PR and several other PR that I am also working on, I don't seem to have write access to this repository, so I was never able to merge branches into autoware.universe. I will not update main branch into current branch this time, could you please help me "squash and merge" this branch？ It's been my pleasure working with you.

**Comment by technolojin** [comment]
@tby-udel 
I did not know that. I confirmed your PR and it is merged. Thank you for your contribution!

**Comment by technolojin** [comment]
https://autowarefoundation.github.io/autoware.universe/main/perception/autoware_multi_object_tracker/ shows error
Screenshot from 2024-07-24 15-07-02

@tby-udel Can you find the reason of error?

**Comment by tby-udel** [comment]
> https://autowarefoundation.github.io/autoware.universe/main/perception/autoware_multi_object_tracker/ shows error Screenshot from 2024-07-24 15-07-02
> 
> @tby-udel Can you find the reason of error?

I think this is caused by some parameters in schema files are lack of certain default values, I will update this multi-object-tracker branch today. Can I still use push and update on this branch?

**Pull request merged**
